### PR TITLE
User + Rooms

### PR DESCRIPTION
**Pull request by zellwk**
Bad commit message. Added functionality for: 
1. Get user
2. Create user
3. Join room
4. Leave room

**todo:**
1. Message room (add message to database)

@cbas Can you help look through this PR? :)

### CODE REVIEW

**Comment by zellwk** [comment]
@sevresbabylone take a look if you want a stab at it too!

File: index.js [hunk 1]
```diff
@@ -40,21 +118,20 @@ io.on('connection', function(socket) {
...
   socket.on('chat message', function(msg) {
-    // save message to database
     msg.message = md.toHTML(msg.message);
     const message = new Message(msg);
     message.save(function(err) {
       if (err) return console.error(err);
     });
-    console.log(`Message: ${msg.message}`);
-    // send message to all clients
+    // console.log(`Message: ${msg.message}`);
```
Comment (sebdeckers) [inline index.js]: Clean up logs

File: index.js [hunk 2]
```diff
@@ -40,21 +118,20 @@ io.on('connection', function(socket) {
...
     io.emit('chat message', msg);
   });
+
   socket.on('typing', function(input) {
-      io.emit('typing', input);
-      console.log(input);
-    });
-  // socket.on('isTyping', input {})
+    io.emit('typing', input);
+    // console.log(input);
```
Comment (sebdeckers) [inline index.js]: Clean up logs

File: collections/room.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+
+const roomSchema = new mongoose.Schema({
+  name: String,
+  messages: [
```
Comment (sebdeckers) [inline collections/room.js:5]: I believe `messages` can use a nested `mongoose.Schema` instance.

``` js
var childSchema = new Schema({ name: 'string' });

var parentSchema = new Schema({
  child: childSchema
});
```

http://mongoosejs.com/docs/subdocs.html#altsyntax

File: collections/message.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const mongoose = require('mongoose');
+
+const Message = mongoose.model('Message', {
```
Comment (sebdeckers) [inline collections/message.js:3]: Does this require a `mongoose.Schema` to be passed to the `Mongoose.model`?

http://mongoosejs.com/docs/models.html

File: index.js [hunk 5]
```diff
@@ -32,7 +30,87 @@ io.use(socketioJwt.authorize({
...
+  });
+
+  socket.on('create user', req => {
+    var user = new User({
+      username: req.username,
+      rooms: req.rooms || []
+    });
+
+    user.save()
+    .then(data => {
```
Comment (sebdeckers) [inline index.js]: Indentation +2

File: index.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+const chalk = require('chalk');
```
Comment (sebdeckers) [inline index.js]: Using `chalk` anywhere?

File: index.js [hunk 7]
```diff
@@ -32,7 +30,87 @@ io.use(socketioJwt.authorize({
...
+      }
+    );
+
+    socket.join(data.room, function(err) {
+      socket.emit('join room', err, `Joined ${data.room}`);
+    });
+  });
+
+  socket.on('leave room', function(data) {
+    console.log('leaving room');
```
Comment (sebdeckers) [inline index.js]: Check out the [debug](https://www.npmjs.com/package/debug) module for this system event logging.

File: package.json [hunk 8]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "forever index.js",
+    "dev": "nodemon index.js",
     "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (sebdeckers) [inline package.json]: Would love to see some form of testing and/or linting to help keep the code sane. Set up [standard](https://github.com/feross/standard) maybe?

File: collections/room.js (hunk 3 above)
Comment (zellwk) [inline collections/room.js:5]: Sounds good. will check it out.

File: package.json [hunk 9]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "start": "forever index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "dev": "nodemon index.js",
```
Comment (zellwk) [inline package.json:8]: Linting set up. Testing will come slightly :)

File: index.js (hunk 6 above)
Comment (zellwk) [inline index.js]: Commented out chalk. Used for dev logging previously.

File: index.js (hunk 5 above)
Comment (zellwk) [inline index.js]: fixed

File: collections/message.js (hunk 4 above)
Comment (zellwk) [inline collections/message.js:3]: Legacy code from David and Jesstern's app. Will remove when rooms are fully implemented

File: collections/room.js (hunk 3 above)
Comment (zellwk) [inline collections/room.js:5]: screen shot 2016-01-12 at 11 14 44 am

Already declared with alternate syntax. Currently don't see a need to access subdoc Schema instance, so I'll leave it like this.

**Comment by sebdeckers** [comment]
LGTM :+1:

**Pull request merged**